lib/puppet/provider/network_vlan/cisco_nexus.rb

```diff
@@ -0,0 +1,44 @@
+require 'puppet/resource_api/simple_provider'
+require 'cisco_node_utils'
+
+# Implementation for the network_vlan type using the Resource API.
+class Puppet::Provider::NetworkVlan::CiscoNexus < Puppet::ResourceApi::SimpleProvider
+  def get(_context)
+    vlan_instances = []
+    Cisco::Vlan.vlans.each do |vlan_id, v|
+      vlan = {
+        ensure:    'present',
+        id:        vlan_id,
+        vlan_name: v.send(:vlan_name),
+        shutdown:  v.send(:shutdown),
+      }
+      vlan_instances << vlan
+    end
+    vlan_instances
+  end
+
+  def create_update(name, should, create_bool)
```

Comment:
Before performing the create/update, might be best to validate the `id` attribute, as the type says `string` but the device only support numerical attributes. Currently it will hit node_utils and throw an argument exception, by performing the validate before this happens it means we provide feedback sooner, and do not make unnecessary API calls. 

Something like: https://github.com/puppetlabs/cisco-network-puppet-module/pull/16/files#diff-d56ec5a9717184d99ed246168591b8ceR56

Reply:
I'll validate that the string is a numeric value, though our call to node utils effectively does this too.

I'm not keen on placing any restrictions that aren't already there, there may be different variants of model / OS / config that do not have this restriction. I'm aiming for feature parity, this type of validation does not already exist.

Reply:
Yea the name limit isn't a huge problem, validating that the id string is numerical is more beneficial though, as it will take few milliseconds for the provider to validate, whereas if we let node_utils handle the validation then it could be a few seconds as it does a lot more in background such as getting all running configs etc.